src/components/auth/GoogleAuthButton.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState } from 'react';
+import { auth } from '../../lib/firebase';
+import { signInWithPopup, GoogleAuthProvider } from 'firebase/auth';
+
+interface GoogleAuthButtonProps {
+  onSuccess?: () => void;
+  onError?: (error: any) => void;
+  label?: string;
+  className?: string;
+}
+
+export const GoogleAuthButton: React.FC<GoogleAuthButtonProps> = ({
+  onSuccess,
+  onError,
+  label = 'Continuar con Google',
+  className = 'w-full py-2 bg-white/10 hover:bg-white/20 text-white rounded flex items-center justify-center',
+}) => {
+  const [loading, setLoading] = useState(false);
+
+  const handleGoogleLogin = async () => {
+    console.log('GoogleAuthButton: Inicio del proceso de login');
+
+    if (loading) {
+      console.log(
+        'GoogleAuthButton: Ya hay un proceso de login en curso, ignorando clic'
+      );
+      return;
+    }
+
+    setLoading(true);
+
+    try {
+      console.log('GoogleAuthButton: Verificando disponibilidad de auth');
+      if (!auth) {
+        throw new Error('Firebase no está inicializado');
+      }
+
+      console.log('GoogleAuthButton: Creando provider de Google');
+      const provider = new GoogleAuthProvider();
+      provider.setCustomParameters({ prompt: 'select_account' });
+
+      // Guardar URL de retorno antes de iniciar el popup
+      const currentUrl = window.location.href;
+      localStorage.setItem('auth_return_url', currentUrl);
+
+      console.log('GoogleAuthButton: Llamando a signInWithPopup');
+      const result = await signInWithPopup(auth, provider);
+      console.log('GoogleAuthButton: Login exitoso', result.user?.email);
+
+      if (result.user) {
+        try {
+          // Importar y llamar a la función
+          const { verifyOrCreateUser } = await import(
+            '../../services/userService'
+          );
+          await verifyOrCreateUser();
+          console.log('Usuario verificado en Strapi después del login');
+        } catch (verifyError) {
+          console.error('Error al verificar usuario en Strapi:', verifyError);
+          // No bloquear el flujo si falla
+        }
+      }
+      if (onSuccess) {
+        console.log('GoogleAuthButton: Llamando callback de éxito');
+        onSuccess();
+        // Guardar un flag en localStorage para saber que acabamos de autenticar
+        localStorage.setItem('just_authenticated', 'true');
+        localStorage.setItem('auth_user_email', result.user?.email || '');
+        window.location.href = window.location.href;
+      }
+
+      // FORZAR RECARGA DE LA PÁGINA DESPUÉS DE LOGIN EXITOSO
+      console.log('GoogleAuthButton: Recargando página para aplicar sesión...');
+      window.location.reload();
+    } catch (error) {
+      // Resto del código de manejo de errores...
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  return (
+    <button
+      onClick={handleGoogleLogin}
+      disabled={loading}
+      className={className}
```

Comment:
Aca quedaria algo asi (chequear que es fijo y que no pero creo que todo es fijo en este caso):
```suggestion
      className={`${className} w-full py-2 bg-white/10 hover:bg-white/20 text-white rounded flex items-center justify-center`}
```